modules/tasks/tasks/src/main/java/org/motechproject/tasks/constants/EventDataKeys.java

```diff
@@ -29,4 +29,7 @@ private EventDataKeys() {
     public static final String REPEAT_INTERVAL_TIME = "repeatIntervalInSeconds";
     public static final String JOB_SUBJECT = "jobSubject";
     public static final String TASK_ID = "task_ID";
+    public static final String TASK_ACTIVITY_ID = "task_activity_ID";
```

Comment:
I don't really think there's any way to avoid this. MotechEvent is basically everything that a MOTECH instance receives so that metadata must live somewhere in the MotechEvent, so we know which task/activity the event is connected to. Metadata map sounds like a good plan, so I can proceed with that.